### PR TITLE
Grants publicuser when creating a new db user/role

### LINKED ISSUE #5019: Org users accessing other org users public tables
Users in the same organization should be able to query other users' public tables, right now it's not possible because users are not granted with the same permission `publicuser` has.

As an initial fix for this the best way is to `grant publicuser to :username;`.

### PR DESCRIPTION
**Pull request by rochoa**
This takes advantage of `publicuser` already being granted with usage permissions on user schemas.

TODO:
 - [x] rake task to grant publicuser to existent users?

This shouldn't be handled like this if we had something like https://github.com/CartoDB/cartodb-postgresql/issues/111

This will fix #5019

### CODE REVIEW

**Comment by rochoa** [comment]
cc/ @Kartones @javisantana @zenitraM 

What do you think?

**Comment by Kartones** [comment]
This also should mean  that scripts that provision a new DB machine (or the OS install's `carto_db_xxx` DB) should also setup `publicuser` before any user is created no? upon initial init I mean.

All that simplifies code, more than welcome!

**Comment by rochoa** [comment]
See https://github.com/CartoDB/cartodb/blob/master/lib/tasks/setup.rake#L93-L108 + https://github.com/CartoDB/cartodb/blob/master/lib/tasks/setup.rake#L21

About db provision, check https://github.com/CartoDB/cartodb-postgresql/issues/111

**Comment by Kartones** [comment]
Then marvellous :dancer:

**Comment by rochoa** [comment]
Ops, I missed the last two comments :disappointed:

**Comment by rochoa** [comment]
bump :exclamation:

File: lib/tasks/db_maintenance.rake [hunk 1]
```diff
@@ -317,6 +317,26 @@ namespace :cartodb do
...
+        user = Carto::User.where(id: user_id).first
+        # already granted users will raise a NOTICE
+        grant_query = "GRANT publicuser to \"#{user.database_username}\""
+        conn = user.in_database(as: :cluster_admin)
+        begin
+          conn.execute(grant_query)
+        rescue => e
+          log("Failed to execute `#{grant_query}`", :grant_publicuser_to_all_users.to_s, user.database_host)
+        ensure
+          conn.close unless conn.nil?
```
Comment (Kartones) [inline lib/tasks/db_maintenance.rake]: nice, we can try this, if works as it should it shouldn't leave open connections in the pooler (and we won't need to use the old model and forcibly close all user connections)

**Comment by Kartones** [comment]
so... for me is a :+1:

**Pull request merged**